src/lightning_app/core/work.py

```diff
@@ -629,3 +632,45 @@ def apply_flow_delta(self, delta: Delta):
                 property_object.fset(self, value)
             else:
                 self._default_setattr(name, value)
+
+    def configure_layout(self) -> Union[None, str, "Frontend"]:
+        """Configure the UI of this LightningWork.
+
+        You can either
+
+        1.  Return a single :class:`~lightning_app.frontend.frontend.Frontend` object to serve a user interface
+            for this Work.
+        2.  Return a string containing a URL to act as the user interface for this Work.
+        3.  Return ``None`` to indicate that this Work doesn't currently have a user interface.
+
+        **Example:** Serve a static directory (with at least a file index.html inside).
+
+        .. code-block:: python
+
+            from lightning_app.frontend import StaticWebFrontend
+
+
+            class Work(LightningWork):
+                def configure_layout(self):
+                    return StaticWebFrontend("path/to/folder/to/serve")
+
+        **Example:** Arrange the UI of my children in tabs (default UI by Lightning).
+
+        .. code-block:: python
+
+            class Work(LightningWork):
+                def configure_layout(self):
+                    return [
+                        dict(name="First Tab", content=self.child0),
+                        dict(name="Second Tab", content=self.child1),
+                        dict(name="Lightning", content="https://lightning.ai"),
+                    ]
+
+        If you don't implement ``configure_layout``, Lightning will use ``self.url``.
```

Comment:
Wouldn't this negate the original goal of figuring out whether a work has a UI or not

> Primary: We need a way for the framework to know if a given app is headless. Since many apps now rely on the implicit configure_layout implemented in the _RootFlow, we need a way to know if a work is expected to host a UI so we know whether or not to configure a layout there.

Reply:
In a way, the contract now is:
- if you manually point a flow to a work, we assume it has a UI (even if it doesn't have a `configure_layout`)
- if you pass a work directly to `LightningApp`, the root flow will only point to it if it overrides `configure_layout`

So there is an edge case there if your work has a UI, doesn't override `configure_layout` and is passed on its own to `LightningApp`, but this was the only way to preserve backwards compatibility